Source/SGDLib/SimpleDistGradAggregator.h

```diff
@@ -326,36 +305,100 @@ class SimpleDistGradAggregator : public IDistGradAggregator<ElemType>
         if (!m_mpi->IsMainNode())
             m_mpi->Isend(headerCPU, headerCPU->Size(), MPI_CHAR, m_mpi->MainNodeRank(), numGradMatrices, &sendHeaderRequest) || MpiFail("MPI_Isend");
 
+
+        // New aggregation pipeline for non-GDR
         // Perform async allreduce on the gradient data
         std::vector<MPI_Request> allReduceRequests;
-        if (!m_nccl.IsSupported())
+        size_t gpuToCpuIndex = 0;
+        size_t cpuToGpuIndex = 0;
+        size_t allReduceIndex = 0;
+        // non-GDR && GPU
+        if ((m_mpi->UseGpuGdr() == 0) && (deviceId != CPUDEVICE))
         {
-            size_t allReduceIndex = 0;
+            Matrix<ElemType>* gpuCopyBuffer = m_aggregationBuffer.get();
+
             ElemType* reductionBuffer;
+            size_t j;
```

Comment:
>j [](start = 19, length = 1)

`j` is not initialized but used in `m_mpi->AllReduce`. Also can you rename `j` to some meaningful name?